### PR TITLE
feat: add OpenTelemetry feature into Gravitee Node

### PR DESCRIPTION
**Pull request by guillaumelamirand**
**Issue**

https://gravitee.atlassian.net/browse/ARCHI-401

**Description**

The objective is to provide a generic way for Gravitee Application to manage tracing based on OpenTelemetry specification.

**Additional context**

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `7.0.0-archi-401-opentelemetry-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/node/gravitee-node/7.0.0-archi-401-opentelemetry-SNAPSHOT/gravitee-node-7.0.0-archi-401-opentelemetry-SNAPSHOT.zip)

### CODE REVIEW

**Review by jhaeyaert — CHANGES_REQUESTED** [review]

File: gravitee-node-opentelemetry/src/main/java/io/gravitee/node/opentelemetry/OpenTelemetryFactory.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                .builder()
+                .setPropagators(
+                    ContextPropagators.create(
+                        TextMapPropagator.composite(W3CTraceContextPropagator.getInstance(), W3CBaggagePropagator.getInstance())
+                    )
+                );
+
+            SdkTracerProvider tracerProvider = SdkTracerProvider
+                .builder()
+                .addSpanProcessor(BatchSpanProcessor.builder(exporterFactory.createSpanExporter()).build())
```
[the remark this answers is not in the document]
Comment (guillaumelamirand) [reply gravitee-node-opentelemetry/src/main/java/io/gravitee/node/opentelemetry/OpenTelemetryFactory.java:80]: See https://gravitee.atlassian.net/browse/ARCHI-428

File: gravitee-node-opentelemetry/src/main/java/io/gravitee/node/opentelemetry/exporter/ExporterFactory.java [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        Protocol protocol = getProtocol();
+        if (protocol == Protocol.HTTP_PROTOBUF || protocol == Protocol.HTTP) {
+            return createHttpSpanExporter(tracesUri, protocol);
+        } else {
+            return createGrpcSpanExporter(tracesUri);
+        }
+    }
+
+    private SpanExporter createGrpcSpanExporter(final URI tracesUri) {
+        return new VertxGrpcSpanExporter(
```
[the remark this answers is not in the document]
Comment (guillaumelamirand) [reply gravitee-node-opentelemetry/src/main/java/io/gravitee/node/opentelemetry/exporter/ExporterFactory.java:70]: See https://gravitee.atlassian.net/browse/ARCHI-427

**Review by jhaeyaert — APPROVED** [review]

**Pull request merged**